commit 6447828875b7d768e4ef0f58765b4bd4e16bcf18
Author: Nikolay Borisov <nik.borisov@suse.com>
Date:   Tue Feb 25 16:33:48 2025 +0200

    x86/mce/inject: Remove call to mce_notify_irq()
    
    The call to mce_notify_irq() has been there since the initial version of
    the soft inject mce machinery, introduced in
    
      ea149b36c7f5 ("x86, mce: add basic error injection infrastructure").
    
    At that time it was functional since injecting an MCE resulted in the
    following call chain:
    
      raise_mce()
        ->machine_check_poll()
            ->mce_log() - sets notfiy_user_bit
      ->mce_notify_user() (current mce_notify_irq) consumed the bit and called the
      usermode helper.
    
    However, with the introduction of
    
      011d82611172 ("RAS: Add a Corrected Errors Collector")
    
    the code got moved around and the usermode helper began to be called via the
    early notifier mce_first_notifier() rendering the call in raise_local()
    defunct as the mce_need_notify bit (ex notify_user) is only being set from the
    early notifier.
    
    Remove the noop call and make mce_notify_irq() static.
    
    No functional changes.
    
    Signed-off-by: Nikolay Borisov <nik.borisov@suse.com>
    Signed-off-by: Borislav Petkov (AMD) <bp@alien8.de>
    Link: https://lore.kernel.org/r/20250225143348.268469-1-nik.borisov@suse.com

diff --git a/arch/x86/kernel/cpu/mce/inject.c b/arch/x86/kernel/cpu/mce/inject.c
index 313fe682db33..06e3cf7229ce 100644
--- a/arch/x86/kernel/cpu/mce/inject.c
+++ b/arch/x86/kernel/cpu/mce/inject.c
@@ -229,7 +229,6 @@ static int raise_local(void)
 	} else if (m->status) {
 		pr_info("Starting machine check poll CPU %d\n", cpu);
 		raise_poll(m);
-		mce_notify_irq();
 		pr_info("Machine check poll done on CPU %d\n", cpu);
 	} else
 		m->finished = 0;